ecs-cli/modules/clients/aws/secretsmanager/client.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2015-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package secretsmanager
+
+import (
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/clients"
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/config"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/aws/aws-sdk-go/service/secretsmanager/secretsmanageriface"
+)
+
+// SMClient defines methods for interacting with the SecretsManagerAPI interface
+type SMClient interface {
+	CreateSecret(secretsmanager.CreateSecretInput) (*secretsmanager.CreateSecretOutput, error)
+	ListSecrets(*string) (*secretsmanager.ListSecretsOutput, error)
+}
+
+type secretsManagerClient struct {
+	client secretsmanageriface.SecretsManagerAPI
+	config *config.CommandConfig
+}
+
+// NewSecretsManagerClient creates an instance of an secretsManagerClient
+func NewSecretsManagerClient(config *config.CommandConfig) SMClient {
+	client := secretsmanager.New(config.Session)
+	client.Handlers.Build.PushBackNamed(clients.CustomUserAgentHandler())
+
+	return newClient(config, client)
+}
+
+func newClient(config *config.CommandConfig, client secretsmanageriface.SecretsManagerAPI) SMClient {
+	return &secretsManagerClient{
+		client: client,
+		config: config,
+	}
+}
+
+func (c *secretsManagerClient) CreateSecret(input secretsmanager.CreateSecretInput) (*secretsmanager.CreateSecretOutput, error) {
+	output, err := c.client.CreateSecret(&input)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return output, nil
+}
+
+func (c *secretsManagerClient) ListSecrets(nextToken *string) (*secretsmanager.ListSecretsOutput, error) {
+	request := secretsmanager.ListSecretsInput{
+		NextToken: nextToken,
```

Comment:
In the AWS SDK go, all of the List API methods have a version that doesn't use a next token and instead takes an input function and iterates over the pages of the API output. You can find it below the method that you're using in the docs (`ListSecretsPages`): https://docs.aws.amazon.com/sdk-for-go/api/service/secretsmanager/#SecretsManager.ListSecrets

I strongly recommend using this in your client because it will save you from having to write the same looping code that the SDK already provides in future PRs.

Reply:
I wasn't set on doing any filtering in the first iteration. However, def good to know `ListSecretsPages` exists. I'm leaning towards adding both for now so that we can try diff approaches in later PRs; we can always remove one if it ends up unused.

Reply:
If you plan on using more than one page of List results, I'd suggest adding it.

Reply:
Hmm I don't us want to be constrained to 1 page of results, but there doesn't seem to be a way with `ListSecretsPages` to gate how many pages we look through before pausing or taking other action. I'm uncomfortable executing an open-ended search through up to 40,000 secrets (or more if acct limits are variable) before moving on. My preference would be to page through a certain # of pages or after a certain number of hits, and either msg or (if we want to get fancy) offer to take action on the next page of results. 

The intention isn't to use ListSecrets{Pages}() to search for a specific resource and then exit, so this seems to be a different use case than the namespace query in https://github.com/aws/amazon-ecs-cli/pull/565. So, I don't think I'm going to add it just yet, but will take some time to mull over where it might help.

Reply:
Its up to you if you want to add it or not.

> but there doesn't seem to be a way with ListSecretsPages to gate how many pages we look through before pausing or taking other action

You can only look through X number of pages with ListSecretsPages by using a counter variable to track the number of pages you have seen. If needed I can show you how this code would look.

Reply:
Or actually I can just paste the example from the docs link with I already set:

```
// Example iterating over at most 3 pages of a ListSecrets operation.
pageNum := 0
err := client.ListSecretsPages(params,
    func(page *ListSecretsOutput, lastPage bool) bool {
        pageNum++
        fmt.Println(page)
        return pageNum <= 3
    })
```

Reply:
Ah, that example does help! My thinking was too focused on the method itself (vs. what we might pass in), I guess. This addresses my concern re: open ended pagination.